src/menu/MenuSongSelection.as

```diff
@@ -991,9 +991,16 @@ package menu
                 songInfoTitle.width = 164;
                 info.addChild(songInfoTitle);
 
+                // Refresh button
+                var refreshBtn:BoxButton = new BoxButton(19, 19, "R");
+                refreshBtn.x = info.width - refreshBtn.width + 2;
+                refreshBtn.y = 2;
+                refreshBtn.addEventListener(MouseEvent.CLICK, refreshHighscoresClick);
+                info.addChild(refreshBtn);
+
                 infoRanks = _gvars.activeUser.getLevelRank(songDetails);
                 var highscores:Object = _gvars.getHighscores(songDetails.level);
-                if (highscores)
+                if (highscores && highscores[1])
```

Comment:
What's the purpose of this exactly?

Reply:
When you finish a song, go to the highscore popup, go to any page other than page 1 and click 'Refresh', only the scores for that page would be loaded. Then, when you go back to the song selection menu and look at that song's highscore info pane, if '&& highscores[1]' isn't included in the check, only your rank and raw score would be displayed as shown in the image below.

![no_highscores](https://user-images.githubusercontent.com/63449095/84345401-d4889900-abdf-11ea-96f0-b022911b514b.png)

This is because the top 5 scores aren't loaded and other scores other than them are. In this edge case, we need to check whether the top 1 highscore is loaded. If it's loaded, we could display them; if not, we need to load them.

Reply:
In that event, I would change to that line to be:
`if (highscores && highscores["1"]) // Check for rank 1 entry.`
As without the quotes on 1 it looks like an array access, since objects keys are normally strings, just to make it clearer in future reading.

Reply:
Oh, I didn't know about that. I've changed that line.